### ISSUE TITLE
This model is very difficult to train. What training skills are there？

### ISSUE THREAD

**Issue by Z-Xiong** [eugenelet/Meta-rPPG#8]
Thanks a lot for releasing the code of Meta-rPPG paper!
We use [the Deap dataset](http://www.eecs.qmul.ac.uk/mmv/datasets/deap/) train Meta-rPPG model, but the model is difficult to converge. Do you have any training skills?
Thaks a lot!

**Comment by kite323** [comment]
> Thanks a lot for releasing the code of Meta-rPPG paper!
> We use [the Deap dataset](http://www.eecs.qmul.ac.uk/mmv/datasets/deap/) train Meta-rPPG model, but the model is difficult to converge. Do you have any training skills?
> Thaks a lot!

Hello, How do you process the DEAP dataset?

**Comment by eugenelet** [comment]
Hi,

A simple guideline is to pay attention to the partitioning of meta-training set into query and support sets. The rest of the training is straightforward and can be directly derived using our released code. Good luck!

Eugene

**Comment by Z-Xiong** [comment]
> > Thanks a lot for releasing the code of Meta-rPPG paper!
> > We use [the Deap dataset](http://www.eecs.qmul.ac.uk/mmv/datasets/deap/) train Meta-rPPG model, but the model is difficult to converge. Do you have any training skills?
> > Thaks a lot!
> 
> Hello, How do you process the DEAP dataset?

I use PFLD to extract facial key points, and use the key points to generate facial ROI and mask. Then I downsampling the image and plethysmograph signal to 30fps.
However, members of our group found that DEAPS dataset is difficult to converge. I recommend that you use UBFC dataset. Its video data and PPG signal are better, so you can use them directly without preprocessing.

**Comment by Z-Xiong** [comment]
> Hi,
> 
> A simple guideline is to pay attention to the partitioning of meta-training set into query and support sets. The rest of the training is straightforward and can be directly derived using our released code. Good luck!
> 
> Eugene


I use the data example “exampl.pth” provided by you to train 30 iterations with default parameters. The loss is as follows. It seems that there is no convergence. I drew the prediction label and the real label, and found that the network didn't seem to be trained. Is it because there are too few samples?
```----------- Networks initialized -------------
Total number of parameters : 0.380 M
---------------------end----------------------
dataset [rPPGDataset-train] was created
dataset [rPPGDataset-test] was created
Data Size: 650 ||||| Batch Size: 3 ||||| initial lr: 0.001000
Epoch 1/30 ||||| Time: 9 sec ||||| Lr: 0.0009141 ||||| Loss: 27.127/27.025
Epoch 2/30 ||||| Time: 10 sec ||||| Lr: 0.0006891 ||||| Loss: 26.496/26.333
Epoch 3/30 ||||| Time: 10 sec ||||| Lr: 0.0004109 ||||| Loss: 25.989/25.749
Epoch 4/30 ||||| Time: 9 sec ||||| Lr: 0.0001859 ||||| Loss: 25.520/25.350
Epoch 5/30 ||||| Time: 8 sec ||||| Lr: 0.0001000 ||||| Loss: 25.441/25.150
Epoch 6/30 ||||| Time: 8 sec ||||| Lr: 0.0001859 ||||| Loss: 25.147/25.039
Epoch 7/30 ||||| Time: 8 sec ||||| Lr: 0.0004109 ||||| Loss: 24.989/24.829
Epoch 8/30 ||||| Time: 8 sec ||||| Lr: 0.0006891 ||||| Loss: 24.463/24.353
Epoch 9/30 ||||| Time: 12 sec ||||| Lr: 0.0009141 ||||| Loss: 23.767/23.556
Epoch 10/30 ||||| Time: 9 sec ||||| Lr: 0.0010000 ||||| Loss: 22.900/22.494
Epoch 11/30 ||||| Time: 9 sec ||||| Lr: 0.0009141 ||||| Loss: 22.128/21.585
Epoch 12/30 ||||| Time: 8 sec ||||| Lr: 0.0006891 ||||| Loss: 21.419/21.028
Epoch 13/30 ||||| Time: 8 sec ||||| Lr: 0.0004109 ||||| Loss: 21.424/20.806
Epoch 14/30 ||||| Time: 8 sec ||||| Lr: 0.0001859 ||||| Loss: 21.671/20.743
Epoch 15/30 ||||| Time: 12 sec ||||| Lr: 0.0001000 ||||| Loss: 20.855/20.712
Epoch 16/30 ||||| Time: 9 sec ||||| Lr: 0.0001859 ||||| Loss: 21.125/20.700
Epoch 17/30 ||||| Time: 10 sec ||||| Lr: 0.0004109 ||||| Loss: 21.850/20.678
Epoch 18/30 ||||| Time: 8 sec ||||| Lr: 0.0006891 ||||| Loss: 21.629/20.645
Epoch 19/30 ||||| Time: 9 sec ||||| Lr: 0.0009141 ||||| Loss: 21.301/20.597
Epoch 20/30 ||||| Time: 8 sec ||||| Lr: 0.0010000 ||||| Loss: 20.735/20.559
Epoch 21/30 ||||| Time: 8 sec ||||| Lr: 0.0009141 ||||| Loss: 21.225/20.634
Epoch 22/30 ||||| Time: 9 sec ||||| Lr: 0.0006891 ||||| Loss: 21.797/20.564
Epoch 23/30 ||||| Time: 9 sec ||||| Lr: 0.0004109 ||||| Loss: 21.015/20.575
Epoch 24/30 ||||| Time: 8 sec ||||| Lr: 0.0001859 ||||| Loss: 21.235/20.549
Epoch 25/30 ||||| Time: 8 sec ||||| Lr: 0.0001000 ||||| Loss: 21.467/20.539
Epoch 26/30 ||||| Time: 8 sec ||||| Lr: 0.0001859 ||||| Loss: 21.035/20.538
Epoch 27/30 ||||| Time: 10 sec ||||| Lr: 0.0004109 ||||| Loss: 21.793/20.538
Epoch 28/30 ||||| Time: 11 sec ||||| Lr: 0.0006891 ||||| Loss: 21.228/20.547
Epoch 29/30 ||||| Time: 11 sec ||||| Lr: 0.0009141 ||||| Loss: 21.193/20.504
Epoch 30/30 ||||| Time: 9 sec ||||| Lr: 0.0010000 ||||| Loss: 21.438/20.281

Process finished with exit code 0```

![Uploading 图片.png…]()

**Comment by Z-Xiong** [comment]
> Hi,
> 
> A simple guideline is to pay attention to the partitioning of meta-training set into query and support sets. The rest of the training is straightforward and can be directly derived using our released code. Good luck!
> 
> Eugene
I use the data example “example.pth” provided by you to train 30 iterations with default parameters. The loss is as follows. It seems that there is no convergence. I drew the prediction label and the real label, and found that the network didn't seem to be trained. Is it because there are too few samples?
```----------- Networks initialized -------------
Total number of parameters : 0.380 M
---------------------end----------------------
dataset [rPPGDataset-train] was created
dataset [rPPGDataset-test] was created
Data Size: 650 ||||| Batch Size: 3 ||||| initial lr: 0.001000
Epoch 1/30 ||||| Time: 9 sec ||||| Lr: 0.0009141 ||||| Loss: 27.127/27.025
Epoch 2/30 ||||| Time: 10 sec ||||| Lr: 0.0006891 ||||| Loss: 26.496/26.333
Epoch 3/30 ||||| Time: 10 sec ||||| Lr: 0.0004109 ||||| Loss: 25.989/25.749
Epoch 4/30 ||||| Time: 9 sec ||||| Lr: 0.0001859 ||||| Loss: 25.520/25.350
Epoch 5/30 ||||| Time: 8 sec ||||| Lr: 0.0001000 ||||| Loss: 25.441/25.150
Epoch 6/30 ||||| Time: 8 sec ||||| Lr: 0.0001859 ||||| Loss: 25.147/25.039
Epoch 7/30 ||||| Time: 8 sec ||||| Lr: 0.0004109 ||||| Loss: 24.989/24.829
Epoch 8/30 ||||| Time: 8 sec ||||| Lr: 0.0006891 ||||| Loss: 24.463/24.353
Epoch 9/30 ||||| Time: 12 sec ||||| Lr: 0.0009141 ||||| Loss: 23.767/23.556
Epoch 10/30 ||||| Time: 9 sec ||||| Lr: 0.0010000 ||||| Loss: 22.900/22.494
Epoch 11/30 ||||| Time: 9 sec ||||| Lr: 0.0009141 ||||| Loss: 22.128/21.585
Epoch 12/30 ||||| Time: 8 sec ||||| Lr: 0.0006891 ||||| Loss: 21.419/21.028
Epoch 13/30 ||||| Time: 8 sec ||||| Lr: 0.0004109 ||||| Loss: 21.424/20.806
Epoch 14/30 ||||| Time: 8 sec ||||| Lr: 0.0001859 ||||| Loss: 21.671/20.743
Epoch 15/30 ||||| Time: 12 sec ||||| Lr: 0.0001000 ||||| Loss: 20.855/20.712
Epoch 16/30 ||||| Time: 9 sec ||||| Lr: 0.0001859 ||||| Loss: 21.125/20.700
Epoch 17/30 ||||| Time: 10 sec ||||| Lr: 0.0004109 ||||| Loss: 21.850/20.678
Epoch 18/30 ||||| Time: 8 sec ||||| Lr: 0.0006891 ||||| Loss: 21.629/20.645
Epoch 19/30 ||||| Time: 9 sec ||||| Lr: 0.0009141 ||||| Loss: 21.301/20.597
Epoch 20/30 ||||| Time: 8 sec ||||| Lr: 0.0010000 ||||| Loss: 20.735/20.559
Epoch 21/30 ||||| Time: 8 sec ||||| Lr: 0.0009141 ||||| Loss: 21.225/20.634
Epoch 22/30 ||||| Time: 9 sec ||||| Lr: 0.0006891 ||||| Loss: 21.797/20.564
Epoch 23/30 ||||| Time: 9 sec ||||| Lr: 0.0004109 ||||| Loss: 21.015/20.575
Epoch 24/30 ||||| Time: 8 sec ||||| Lr: 0.0001859 ||||| Loss: 21.235/20.549
Epoch 25/30 ||||| Time: 8 sec ||||| Lr: 0.0001000 ||||| Loss: 21.467/20.539
Epoch 26/30 ||||| Time: 8 sec ||||| Lr: 0.0001859 ||||| Loss: 21.035/20.538
Epoch 27/30 ||||| Time: 10 sec ||||| Lr: 0.0004109 ||||| Loss: 21.793/20.538
Epoch 28/30 ||||| Time: 11 sec ||||| Lr: 0.0006891 ||||| Loss: 21.228/20.547
Epoch 29/30 ||||| Time: 11 sec ||||| Lr: 0.0009141 ||||| Loss: 21.193/20.504
Epoch 30/30 ||||| Time: 9 sec ||||| Lr: 0.0010000 ||||| Loss: 21.438/20.281

Process finished with exit code 0```

![图片](https://user-images.githubusercontent.com/30321226/107142741-e76c8c80-696b-11eb-82e1-5ddac59b8728.png)

**Comment by DaJun666666** [comment]
I also encountered the same problem, did you solve this problem?
